.github/workflows/go-build-release.yaml

```diff
@@ -0,0 +1,77 @@
+name: Go Release Artifact Publisher
+
+on:
+  workflow_call:
+    inputs:
```

Comment:
I hope we can reduce input values using a github action context.
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context